scipy/stats/_qmc.py

```diff
@@ -1036,6 +1084,56 @@ def _random(self, n: IntNumber = 1) -> np.ndarray:
         self.num_generated += n
         return samples
 
+    def _random_oa_lhs(self, n: IntNumber = 1) -> np.ndarray:
+        """Orthogonal array based LHS of strength 2."""
+        p = np.sqrt(n).astype(int)
+        n_row = p**2
+        n_col = p + 1
+
+        primes = primes_from_2_to(p + 2)
```

Comment:
Though this line does not cause an error, the input of `prime_from_2_to(...)` should ideally be `n >= 6`, according to the docstring of the `prime_from_2_to`. This point should be consistent within the code, IMO.

Reply:
Why 6? The minimal n is 4 as the minimal prime is 2. Not sure what you are suggesting here.

Reply:
I meant the docstring at [line 488](https://github.com/tupui/scipy/blob/oa_lhs/scipy/stats/_qmc.py#L488)
```python
def primes_from_2_to(n: int) -> np.ndarray:
    """Prime numbers from 2 to *n*.
    Parameters
    ----------
    n : int
        Sup bound with ``n >= 6``.
    ...
    """
```

Reply:
Ah I see what you mean. The doc of `prime_from_2_to(...)` talks about `n>=6` but in our case it's not an issue as below this value it simply returns `[2, 3]`. This is just an indication to say that the function does something above 6.

Reply:
I will actually put p+1. Does not make sense to have p+2, confusing and I guess I had this before with my wrong "closest prime" logic.

Reply:
> Indeed, the above function returns the correct output for n>=4. For n=3, it returns `[2, 3]`, which is not in the range `2 <= out < n`. However, the docstring says `n >= 6` should be given as input.

Maybe the doc could be improved (but out of scope here, also it's a private function hence with low value IMO). In our case this is not a problem as we are passing the square root of n. Hence the check does properly fail bellow n<4.